### ISSUE TITLE
Bug: I am getting Invalid hook error. I am a beginner. Looking for some help understanding all the possible causes for this error and how to resolve them. I agree that there is lot of documentation around it. But I am looking f someone could help me understand in Layman's language.

### ISSUE THREAD

**Issue by ArpithaSS** [facebook/react#21311]
Error: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://reactjs.org/link/invalid-hook-call for tips about how to debug and fix this problem.

**Comment by eps1lon** [comment]
Support requests filed as GitHub issues often go unanswered. We want you to find the answer you're looking for, so we suggest the following alternatives:

Coding Questions
If you have a coding question related to React and React DOM, it might be better suited for Stack Overflow. It's a great place to browse through frequent questions about using React, as well as ask for help with specific questions.

https://stackoverflow.com/questions/tagged/react

Talk to other React developers
There are many online forums which are a great place for discussion about best practices and application architecture as well as the future of React.

https://reactjs.org/community/support.html
